### PR TITLE
Notifications: move visible component state to global tree

### PR DESCRIPTION
**Pull request by gwwar**
This PR is needed for #12653, where we remove the notifications-panel iframe in Calypso. The notifications panel has keyboard shortcuts. Because the iframe is removed, any overlapping shortcuts in Calypso and the panel will both fire. For example with notifications open, typing <kbd>j</kbd> will move to the next item in both Reader and the Notifications Panel.

screen shot 2017-05-04 at 11 03 40 am

This PR basically moves a visibility boolean to the global redux tree. In a follow up PR, I'll add a check in `lib/keyboard-shorcuts` to not fire off any handlers if the notes panel is open.

### Testing Instructions
- Navigate to http://calypso.localhost:3000/
- Clicking on the Notifications bell opens the notifications panel and fires off a tracks event
- Clicking on the Notifications bell again closes the panel
- The shortcut <kbd>n</kbd> also toggles the panel open and closed
- In the dev console `dispatch( { type: 'NOTIFICATIONS_PANEL_TOGGLE' } )` toggles notifications
- In console `state.ui.isNotificationsOpen` matches the current panel state

cc @Automattic/lannister

### CODE REVIEW

File: client/state/action-types.js [hunk 1]
```diff
@@ -301,6 +301,8 @@ export const MAGIC_LOGIN_SHOW_CHECK_YOUR_EMAIL_PAGE = 'MAGIC_LOGIN_SHOW_CHECK_YO
 export const MAGIC_LOGIN_SHOW_INTERSTITIAL_PAGE = 'MAGIC_LOGIN_SHOW_INTERSTITIAL_PAGE';
 export const MAGIC_LOGIN_SHOW_REQUEST_FORM = 'MAGIC_LOGIN_SHOW_REQUEST_FORM';
 export const MAGIC_LOGIN_SHOW_LINK_EXPIRED = 'MAGIC_LOGIN_SHOW_LINK_EXPIRED';
+export const MASTER_BAR_NOTIFICATIONS_HIDE = 'MASTER_BAR_NOTIFICATIONS_HIDE';
+export const MASTER_BAR_NOTIFICATIONS_SHOW = 'MASTER_BAR_NOTIFICATIONS_SHOW';
```
Comment (dmsnell) [inline client/state/action-types.js]: how about something like `NOTIFICATIONS_PANEL_SHOW` …

File: client/state/selectors/index.js [hunk 2]
```diff
@@ -163,4 +163,5 @@ export isUserSettingUnsaved from './is-user-setting-unsaved';
 export isWordpressUpdateSuccessful from './is-wordpress-update-successful';
 export isValidatingAccountRecoveryKey from './is-validating-account-recovery-key';
 export shouldCloseVideoEditorModal from './should-close-video-editor-modal';
+export shouldShowMasterbarNotifications from './should-show-masterbar-notifications';
```
Comment (dmsnell) [inline client/state/selectors/index.js]: how about `isNotificationsPanelOpen`

File: client/state/selectors/should-show-masterbar-notifications.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * Returns true if masterbar notifications should be shown.
+ *
+ * @param  {Object}  state - Global state tree
+ * @return {Boolean} true  - if notifications should be shown.
+ *
+ */
+export default function shouldShowMasterbarNotifications( state ) {
+	return get( state, 'ui.masterbar.showNotifications', false );
```
Comment (dmsnell) [inline client/state/selectors/should-show-masterbar-notifications.js]: even with a rename I think it's fine to leave this under `ui.masterbar`. with the selector we are at least free to change it later internally without affecting other things

**Comment by gwwar** [comment]
Thanks for the review @dmsnell @rodrigoi !

**Pull request merged**